sycl/include/sycl/ext/oneapi/experimental/invoke_simd.hpp

```diff
@@ -364,6 +364,11 @@ constexpr bool has_ref_ret(Ret (*)(Args...)) {
   return std::is_reference_v<Ret>;
 }
 
+template <typename Ret, typename... Args>
+constexpr bool has_struct_ret(Ret (*)(Args...)) {
```

Comment:
If I remember it correctly, the structures also cannot be passed to invoke_simd as an argument/parameter.
That is why attempt to pass local_accessor by value to invoke_simd doesn't work now.
Please add the checks for parameters as well.

Reply:
good point, thanks. i just pushed a commit adding the argument check as well as rebasing against current intel/llvm